### PR TITLE
Add Image Tags

### LINKED ISSUE #5: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by aarontcote**
Closes #5

### CODE REVIEW

**Review by Santhanaraman — COMMENTED** [review]
This PR looks great, thanks a lot for splitting the controller tests. One minor comment on the commit messages, I think it is always better to split up the refactor and the new feature commits. For example, in this case, we could have had a commit that splits the tests first and then updated the tests for image tags. Other than that left some minor suggestions and questions around tests

File: test/controllers/images/images_controller_test.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 require 'test_helper'
 
 class ImagesControllerTest < ActionDispatch::IntegrationTest
+  IMAGE_URL = 'https://pa.cdn.appfolio.com/appfolio/images/apm-white.svg'.freeze
```
[the remark this answers is not in the document]
Comment (aarontcote) [reply test/controllers/images/images_controller_test.rb:4]: Good point, I'll break it out further.

File: test/controllers/images/images_controller_test.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
...
+
```
[the remark this answers is not in the document]
Comment (aarontcote) [reply test/controllers/images/images_controller_test.rb:5]: Yes, I'll do that.

File: test/controllers/images/images_controller_show_test.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+require 'test_helper'
+
+class ImagesControllerTest < ActionDispatch::IntegrationTest
```
[the remark this answers is not in the document]
Comment (aarontcote) [reply test/controllers/images/images_controller_show_test.rb:3]: I was thinking they need to be the same for them to be merged in Ruby, but I'll rename if it still knows to combine them.

File: db/seeds.rb [hunk 4]
```diff
@@ -1,25 +1,45 @@
...
-  { url: 'https://images.unsplash.com/photo-1552998841-3a61a27d4173' },
-  { url: 'https://images.unsplash.com/photo-1568833450751-fba3c6b2d129' },
-  { url: 'https://images.unsplash.com/photo-1571781565036-d3f759be73e4' },
-  { url: 'https://images.unsplash.com/photo-1580164631075-b3f1304f4051' },
-  { url: 'https://images.unsplash.com/photo-1557342960-7ea3df1d8630' },
-  { url: 'https://images.unsplash.com/photo-1547130542-00f463cf8c0c' },
-  { url: 'https://images.unsplash.com/photo-1559535332-db9971090158' },
-  { url: 'https://images.unsplash.com/photo-1613276017067-ef618366152a' },
-  { url: 'https://images.unsplash.com/photo-1593783162886-d2277f7d5908' }
+  { url: 'https://images.unsplash.com/photo-1565207192414-5ba80622abb8',
```
Comment (brightmer) [inline db/seeds.rb:3]: I would usually ask this to be in a separate commit since it is mixing of test data and production logic.

File: test/controllers/images/images_controller_create_test.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+  test 'create should create an image url with multiple tags' do
+    assert_difference 'Image.count' do
+      post images_path, params: { image: { url: IMAGE_URL,
+                                           tag_list: 'appfolio, goleta, california' } }
+      assert_redirected_to image_path(Image.last)
+    end
+  end
+
+  test 'create should prevent image url from saving' do
```
Comment (brightmer) [inline test/controllers/images/images_controller_create_test.rb:49]: Does it makes sense to change the name to something like 'create should prevent image url from saving with empty string'
This test doesn't have a _state under test_ in the test name like the other tests, i.e. `with multiple tags`.

**Review by brightmer — COMMENTED** [review]
I was following up on your slack message. I see Santhanaraman is reviewing so I'm just leaving some feedback, act on it if you want :)

File: db/seeds.rb (hunk 4 above)
Comment (aarontcote) [reply db/seeds.rb:3]: That's a good idea, I'll break it out.

File: test/controllers/images/images_controller_create_test.rb (hunk 5 above)
Comment (aarontcote) [reply test/controllers/images/images_controller_create_test.rb:49]: That makes sense, I'll update.

**Review by Santhanaraman — APPROVED** [review]
